### DISCUSSION TITLE
Sidebar function in Instrument keyboard layouts

### DISCUSSION THREAD

**Discussion by PaulFreund** [SynthstromAudible/DelugeFirmware#255] [Polls]
I want to make use of the sidebar in the instrument (Isomorphic and In-Key) layouts in keyboard input (this is not for drums). My idea is to show as  white pads with increasing brightness the different velocities and apply the selected to all new pad presses. What do you think?

**Comment by PaulFreund** [comment]
Little update: One of the columns will be for velocity, the other one probably for modwheel :D

**Reply by sichtbeton** [reply]
Chords as shown here:

https://youtu.be/vCpqs-diMoA

**Reply by manysounds** [reply]
Oooooo that is great. Definitely should be a mode-switch option

**Reply by ok-reza** [reply]
definitely velocity makes sense, would love to see chord mode on the other column. obviously if there's a good UX/UI for customizing those columns then that would be great, but velocity and chord would be the most useful since there are knobs that can handle modulations while playing.

**Comment by manysounds** [comment]
While we're at it, another option for the sidebar could be to change scale mode or base key. I've always had a day-dream about playing jazz with a scale mode keyboard and being able to change key/mode on the fly would facilitate that fully. I briefly experimented with doing that through Ableton/Max to a Launchpad.

**Comment by croteaup** [comment]
Why not make each column function selectable:
Press audition pad or mute pad + turn select knob to scroll through different options.
This way we could have 2-off any of the many things that these 2 columns could become: chord pads, mod wheel, pitch wheel, a 3rd or 4th gold knob, velocity, transpose, arpeggio rate, etc.

**Reply by soymonitus** [reply]
My vote is for this. There are so many possibilities, why make it fixed to just one? make them editable between those options:
chord bank, mod wheel, pitch bend wheel, velocity selector, note repeat at different rates

**Reply by sichtbeton** [reply]
Yes this. Having this selectable would open the possibilities up so much.

**Reply by darenager** [reply]
Yes selectable would be great! A chord memory function as well as fixed chords would be nice too!

**Reply by racecube** [reply]
Anyway but please: Extend Clip length/double clip length from keyboard view like in every other view.

**Comment by robmccoll** [comment]
There's still some work to do and plenty of opportunities for improvement, but I think this is getting close to encompassing the ideas discussed here: https://github.com/SynthstromAudible/DelugeFirmware/pull/901

**Comment by m-m-adams** [comment]
Included in 1.1
